bee-node/src/plugins/mqtt/broker.rs

```
@@ -0,0 +1,133 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    plugins::mqtt::{handlers::*, DEFAULT_MAX_INFLIGHT_REQUESTS},
+    storage::StorageBackend,
+};
+
+use bee_runtime::{node::Node, worker::Worker};
+
+use async_trait::async_trait;
+use librumqttd as mqtt;
+use log::*;
+
+use std::{convert::Infallible, thread};
+
+#[derive(Default)]
+pub struct MqttBroker;
+
+#[async_trait]
+impl<N: Node> Worker<N> for MqttBroker
+where
+    N::Backend: StorageBackend,
+{
+    type Config = mqtt::Config;
+    type Error = Infallible;
+
+    async fn start(node: &mut N, config: Self::Config) -> Result<Self, Self::Error> {
+        let mut broker = mqtt::Broker::new(config);
+
+        let mut milestones_latest_tx = broker.link("milestones-latest").expect("linking mqtt sender failed");
```

There's lots of `expect`s here -- is this something that makes sense to describe with an error result or is the panicking fine?

---

hmm right i guess we should return the error here instead